taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -2749,6 +2749,15 @@ LLVMCompiledTask TaskCodeGenLLVM::run_compilation() {
   emit_to_module();
   eliminate_unused_functions();
 
+#ifdef TI_WITH_CUDA
```

Comment:
If you compiles with CUDA but executes on CPU, this is still problematic. Should check taichi::Arch at runtime as well?

Reply:
you're absolutely right lol, thanks!